docs/development.md

```diff
@@ -99,7 +103,21 @@ The dev version of the manifests can be generated with
 
 1. [Install the gcloud cli][gcloud_sdk].
 2. Set project: `gcloud config set project YOUR_PROJECT_NAME`.
-3. Build & push dev images: `make docker-push-dev`.
+3. To build images with custom tags, edit [.bazelrc](./.bazelrc) as follows:
+
+   `build --define=MANAGER_IMAGE_TAG=<YOUR_CUSTOM_TAG_HERE>`
+4. Build & push dev images: `make docker-push-dev`.
+
+#### Building and pushing dev images to custom (non GCR) container registry
+
+1. Login to your container registry using `docker login`.
+
+   E.g. `docker login quay.io`
+2. To build images with custom tags, edit [.bazelrc](./.bazelrc) as follows:
+
+   `build --define=MANAGER_IMAGE_TAG=<YOUR_CUSTOM_TAG_HERE>`
+3. Build dev images: `REGISTRY_DEV=<YOUR_CONTAINER_REGISTRY_HERE> make clean docker-build-dev`
+4. Push your docker images as `docker push <ContainerImage>:<YourTag>`
```

Comment:
this is all achievable with `make docker-push-dev BAZEL_ARGS="--define=MANAGER_IMAGE_TAG=my-manager-test-tag --define=REGISTRY_DEV=registry.example.com"`

Although looking through the bazel file, I'd think `REGISTRY_DEV` and `REGISTRY_STABLE` should both just be `REGISTRY`

https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/cmd/manager/BUILD.bazel#L111
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/cmd/manager/BUILD.bazel#L119

But that's out of scope for this PR

Reply:
yeah. Let's clean that up in a different PR.

Reply:
I think keeping `REGISTRY_DEV` and `REGISTRY_STABLE` makes following the build easier. There isn't much to gain by collapsing the two into one. WDYT?

Reply:
Making the change to collapse the two variables, I kinda confused myself. 🤣